src/pages/rekey-reason/cat-be/__tests__/rekey-reason.cat-be.page.spec.ts

```diff
@@ -0,0 +1,247 @@
+        component.loadingSpinner = loadingController.create();
+
+        component.handleLoadingUI(false);
+
+        expect(component.loadingSpinner).toBeNull();
+      });
+    });
+
+    describe('onShowUploadRekeyModal', () => {
+      const options = { cssClass: 'mes-modal-alert text-zoom-regular' };
+      it('should display an upload modal', () => {
+        component.onShowUploadRekeyModal();
+        expect(modalController.create).toHaveBeenCalledWith('UploadRekeyModal', { retryMode: false }, options);
+      });
+      it('should display an upload modal in retry mode', () => {
+        component.onShowUploadRekeyModal(true);
+        expect(modalController.create).toHaveBeenCalledWith('UploadRekeyModal', { retryMode: true }, options);
+      });
+    });
+
+    describe('handleUploadOutcome', () => {
+      beforeEach(() => {
+        spyOn(component, 'handleLoadingUI');
+        spyOn(component, 'onShowUploadRekeyModal');
+      });
+
+      it('should display the loading spiner when an upload is in progress', () => {
+        const action = new SendCurrentTest();
+        const result: RekeyReasonModel = rekeyReasonReducer(null, action);
+        const uploadStatus = getUploadStatus(result);
+
+        component.handleUploadOutcome(uploadStatus);
+
+        expect(component.handleLoadingUI).toHaveBeenCalledWith(true);
+      });
+      it('should display the retry modal when an upload fails', () => {
+        const action = new SendCurrentTestFailure(false);
+        const result: RekeyReasonModel = rekeyReasonReducer(null, action);
+        const uploadStatus = getUploadStatus(result);
+
+        component.handleUploadOutcome(uploadStatus);
+
+        expect(component.handleLoadingUI).toHaveBeenCalledWith(false);
+        expect(component.onShowUploadRekeyModal).toHaveBeenCalledWith(true);
+
+      });
+      it('should navigate to the next page and not display the retry modal when an upload is a duplicate', () => {
+        const action = new SendCurrentTestFailure(true);
+        const result: RekeyReasonModel = rekeyReasonReducer(null, action);
+        const uploadStatus = getUploadStatus(result);
+
+        component.handleUploadOutcome(uploadStatus);
+
+        expect(component.handleLoadingUI).toHaveBeenCalledWith(false);
+        expect(navContoller.push).toHaveBeenCalledWith(CAT_B.REKEY_UPLOAD_OUTCOME_PAGE);
+        expect(component.onShowUploadRekeyModal).not.toHaveBeenCalled();
+
+      });
+      it('should navigate to next page and not display the retry modal when an upload succeeds', () => {
+        const action = new SendCurrentTestSuccess();
+        const result: RekeyReasonModel = rekeyReasonReducer(null, action);
+        const uploadStatus = getUploadStatus(result);
+
+        component.handleUploadOutcome(uploadStatus);
+
+        expect(component.handleLoadingUI).toHaveBeenCalledWith(false);
+        expect(navContoller.push).toHaveBeenCalledWith(CAT_B.REKEY_UPLOAD_OUTCOME_PAGE);
```

Comment:
CAT B

Reply:
Changing this to use the CAT_BE page (Although at the moment it's the same page since it hasn't been split out)